full/src/test/java/apoc/uuid/UUIDNewProceduresTest.java

```diff
@@ -0,0 +1,494 @@
+        } catch (RuntimeException e) {
+            // then
+            Throwable except = ExceptionUtils.getRootCause(e);
+            assertTrue(except instanceof RuntimeException);
+            assertEquals("No constraint found for label: Wrong, please add the constraint with the following : `CREATE CONSTRAINT ON (wrong:Wrong) ASSERT wrong.foo IS UNIQUE`", except.getMessage());
+            throw e;
+        }
+    }
+
+    //
+    // new test cases
+    //
+
+    @Test
+    public void testUuidShow() {
+        db.executeTransactionally("CREATE CONSTRAINT FOR (n:Show1) REQUIRE n.uuid IS UNIQUE");
+        db.executeTransactionally("CREATE CONSTRAINT FOR (n:Show2) REQUIRE n.uuid IS UNIQUE");
+
+        String label1 = "Show1";
+        String label2 = "Show2";
+        String query = "MATCH (c:TestShow) SET c.count = 1";
+
+        testCall(sysDb, "CALL apoc.uuid.create('neo4j', $name)",
+                map("name", label1),
+                r -> assertEquals(label1, r.get("label")));
+
+        testCall(sysDb, "CALL apoc.uuid.create('neo4j', $name)",
+                map("name", label2),
+                r -> assertEquals(label2, r.get("label")));
+
+        // not updated
+        testResult(sysDb, "CALL apoc.uuid.show('neo4j')",
+                map("query", query, "name", label1),
+                res -> {
+                    Map<String, Object> row = res.next();
+                    assertEquals(label1, row.get("label"));
+                    Map<String, Object> defaultProperties = Map.of(UUID_PROPERTY_KEY, DEFAULT_UUID_PROPERTY,
+                            ADD_TO_SET_LABELS_KEY, DEFAULT_ADD_TO_SET_LABELS);
+                    assertEquals(defaultProperties, row.get("properties"));
+                    row = res.next();
+                    assertEquals(label2, row.get("label"));
+                    assertEquals(defaultProperties, row.get("properties"));
+                    assertFalse(res.hasNext());
+                });
+    }
+
+    @Test
+    public void testInstallUuidInUserDb() {
+        try {
+            testCall(db, "CALL apoc.uuid.create('neo4j', 'AnotherLabel')",
+                    r -> fail("Should fail because of user db execution"));
+        } catch (QueryExecutionException e) {
+            assertThat(e.getMessage(), Matchers.containsString(PROCEDURE_NOT_ROUTED_ERROR));
+        }
+    }
+
+    // TODO - it should be removed/ignored in 5.x, due to Util.validateQuery(..) removal
+    @Test
+    public void testInstallUuidInWrongDb() {
+        try {
+            testCall(sysDb, "CALL apoc.uuid.create('notExistent', 'DbNotExistent')",
+                    r -> fail("Should fail because of database not found"));
+        } catch (QueryExecutionException e) {
+            assertThat(e.getMessage(), Matchers.containsString(DatabaseNotFoundException.class.getName()));
+        }
+    }
+
+    @Test
+    public void testShowUuidInUserDb() {
+        try {
+            testCall(db, "CALL apoc.uuid.show('neo4j')",
```

Comment:
show should be able to run on all databases

Reply:
Changed (and removed `@SystemProcedure`)